readium/navigator/src/main/java/org/readium/r2/navigator/pdf/PdfEngineProvider.kt

```diff
@@ -71,29 +74,28 @@ abstract class PdfDocumentFragment<S: Configurable.Settings> : Fragment() {
     abstract fun goToPageIndex(index: Int, animated: Boolean): Boolean
 
     /**
-     * Current presentation settings for the PDF document.
-     *
-     * WARNING: This API will change when the Presentation API is finalized.
-     * See https://github.com/readium/architecture/pull/164
+     * Current settings for the PDF document.
      */
     abstract var settings: S
 }
 
-@OptIn(ExperimentalReadiumApi::class)
-@PdfSupport
-data class PdfDocumentFragmentInput<S: Configurable.Settings>(
+@ExperimentalReadiumApi
+data class PdfDocumentFragmentInput<S: PdfSettings>(
     val publication: Publication,
     val link: Link,
     val initialPageIndex: Int,
     val settings: S,
     val listener: PdfDocumentFragment.Listener?
 )
 
+
 @ExperimentalReadiumApi
-@PdfSupport
 interface PdfSettings : Configurable.Settings {
 
     val readingProgressionValue: ReadingProgression
 
-    fun update(metadata: Metadata, preferences: Preferences, defaults: Preferences): PdfSettings
+    val scrollValue: Boolean
```

Comment:
- That's a lifecycle issue. We need to be able to provide `readingProgression` before we have a `PdfDocumentFragment`. That's the reason why I decided that `PdfDocumentFragment` had to obey the settings, resolved in `PdfNavigatorViewModel`,  rather than controlling them. We could also deprecate this legacy property from `Navigator `.
- In the testapp, tap navigation is handled in a generic way without dependency to a specific navigator, so I can't go without `isTapNavigationRelevant`. I can make the PDF implementation more straightforward through a specific settings implementation though, indeed.

Reply:
We don't need `isTapNavigationRelevant`, we can just run `EdgeTapNavigation` all the time in the test app like before. Apps can decide to handle it in a finer way.

`VisualNavigator.readingProgression` is still useful for helpers like `EdgeTapNavigation`, so we can't deprecate it. An alternative could be to add `readingProgression(metadata: Metadata, preferences: Preferences)` in `PdfEngineProvider`, but that's only marginally better.

Reply:
I think you'd think differently if you'd tested it. Now that `readingProgression` is either LTR or RTL, tapping on the left or on the right scrolls up or down in scroll mode. I consider this as completely broken.
`EdgeTapNavigation` should probably be modified to accept `readingProgression`, `scroll` and `scrollAxis`. We're now back to our problem : such values are not consistently available across navigators.

Reply:
That's a good point, and I think it goes further than the `Configure` API as `EdgeTapNavigation` should work with any `VisualNavigator`. Maybe `VisualNavigator` should have a `presentationInfo: PresentationInfo`:

```kotlin
data class PresentationInfo(
    val readingProgression: ReadingProgression? = null,
    val scroll: Boolean? = null,
)
```

Then no need to add the conditionals in the Test App, the `EdgeTapNavigation` has everything it needs to determine if we need to move or not.

For most navigators it would be straightforward to produce this data class. For PDF it's a bit more complicated but we can delegate to `PdfEngineProvider` instead of the settings class.

Reply:
I don't like much the name `PresentationInfo`, but the idea sounds good to me.